Jumoo.uSync.BackOffice/Handlers/TemplateHandler.cs

```diff
@@ -149,6 +153,36 @@ private void FileService_SavedTemplate(IFileService sender, Umbraco.Core.Events.
             }
         }
 
+        private Boolean RemoveRootFileIfFound(item) {
+            Boolean removed = false;
+            String foundTemplate = FindTemplate(item.Alias);
+            String GeneratedTemplate = IOHelper.MapPath(string.Format(SystemDirectories.MvcViews + "/{0}.cshtml", item.Alias.ToSafeFileName()));
+            if (!String.IsNullOrEmpty(foundTemplate) && File.ReadAllText(foundTemplate).Equals(item.Content) && File.Exists(GeneratedTemplate) && File.ReadAllText(foundTemplate).Equals(File.ReadAllText(GeneratedTemplate))) {
```

Comment:
Hi 

I am looking at this, and i am not sure if this is the intended behaviour?

my understanding is that when you save a template in Umbraco it writes that file to disk (in the root of the views folder). and the purpose of this function is to find that file, and if it matches one in a sub folder delete it ? 

if however the user has changed something via the back office (say added code, or even a newline) then when they save that file - the one in the root of the views folder will no longer match the one in the sub folder ? (or is umbraco writing to both). 

If they don't match then this line will never fire (for files altered a bit in the back office) and so they won't be deleted? 

I probibly need to do some testing - but would be good to get your thoughts ?

Reply:
Hi Kevin, 

Yes that's indeed the intended behavior but your're definitely right, I haven't thought about the case in which users edit the template files directly via umbraco. Umbraco does not write in both files, only the ones that are accessible on root level in the Views folder. If you move template files in a sub-folder the front-end editor also doesn't show any mark-up. I don't know how this could be solved for now, I'll try to get up with something.

Reply:
I think you will either have to remove the templates tree (in the tree.config file) or maybe put something into a FileSaving Event to stop people saving the templates (but then you have to know if it's a background process like uSync saving the file or the user).